backup: TestDataDriven_online_restore_empty_database failed

backup.TestDataDriven_online_restore_empty_database [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21142069?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21142069?buildTab=artifacts#/) on master @ [92c72ef9f12cffedf63632f5dfbf2cfeee021e70](https://github.com/cockroachdb/cockroach/commits/92c72ef9f12cffedf63632f5dfbf2cfeee021e70):


```
=== RUN   TestDataDriven_online_restore_empty_database
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestDataDriven_online_restore_empty_database2939871997
    test_log_scope.go:82: use -show-logs to present logs inline
    datadriven_test.go:492: Mock HTTP Storage "http://127.0.0.1:40427"
    datadriven_test.go:514: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:3:
          reset test-nodelocal
          ----
    test_server_shim.go:95: cluster virtualization disabled due to issue: #142798 (expected label: C-test-failure)
    test_server_shim.go:679: DRPC is enabled (via testserver factory defaults)
    datadriven_test.go:514: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:6:
          new-cluster name=s1 disable-tenant
          ----
    datadriven_test.go:514: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:10:
          exec-sql
          CREATE DATABASE d;
          ----
    datadriven_test.go:514: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:14:
          exec-sql
          BACKUP INTO 'nodelocal://1/cluster/';
          ----
    datadriven_test.go:514: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:18:
          exec-sql
          RESTORE DATABASE d FROM LATEST IN 'nodelocal://1/cluster/' with EXPERIMENTAL DEFERRED COPY, new_db_name='d2';
          ----
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:22: still running after 10.000481426s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:22: still running after 20.001336091s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:22: still running after 30.001557237s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:22: still running after 40.002592268s
    datadriven_test.go:763: condition failed to evaluate within 45s: from datadriven_test.go:771: latest output: 2
         expected: 1
    datadriven.go:343: 
        fatal during /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18655/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/online-restore-empty-database:22:
        SELECT count(*) FROM [SHOW JOBS] WHERE job_type='RESTORE' and status='succeeded';
    testutils.go:326: no Invalid Descriptors
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestDataDriven_online_restore_empty_database2939871997
--- FAIL: TestDataDriven_online_restore_empty_database (48.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_online_restore_empty_database.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60667

--- a/pkg/backup/testdata/backup-restore/online-restore-empty-database
+++ b/pkg/backup/testdata/backup-restore/online-restore-empty-database
@@ -22,4 +22,4 @@ RESTORE DATABASE d FROM LATEST IN 'nodelocal://1/cluster/' with EXPERIMENTAL DEF
 query-sql retry
 SELECT count(*) FROM [SHOW JOBS] WHERE job_type='RESTORE' and status='succeeded';
 ----
-1
+2

